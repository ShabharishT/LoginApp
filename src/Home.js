import React, {Component} from 'react';

/*
 * Home component welcoming user and
 * providing a logout action
 */
export default class Home extends Component{

    constructor(props) {
        super(props);

        this.handleLogout = this.handleLogout.bind(this);
    }

    handleLogout() {
        this.props.history.push('/login');
    }

    render() {
        const {userName} = this.props.location.state;
        return (
            <div>
                Welcome {userName}!<br/><br/>
                <input type="button" onClick={this.handleLogout} value="Logout"/>
            </div>
        );
    }
}