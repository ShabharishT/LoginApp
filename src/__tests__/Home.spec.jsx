import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home';

describe('Home Component ', () => {

    test ('Successful render', () => {
        const location = {state:'user'};
        const component = renderer.create(<Home location={location}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});