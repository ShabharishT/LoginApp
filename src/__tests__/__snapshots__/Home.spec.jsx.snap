// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component  Successful render 1`] = `
<div>
  Welcome 
  !
  <br />
  <br />
  <input
    onClick={[Function]}
    type="button"
    value="Logout"
  />
</div>
`;
